pages/shelf/[[walletAddress]].vue

```diff
@@ -72,49 +98,96 @@
 
 <script setup lang="ts">
 const { t: $t } = useI18n()
-const { loggedIn: hasLoggedIn } = useUserSession()
+const { loggedIn: hasLoggedIn, user } = useUserSession()
 const localeRoute = useLocaleRoute()
 const bookshelfStore = useBookshelfStore()
+const metadataStore = useMetadataStore()
 const infiniteScrollDetectorElement = useTemplateRef<HTMLLIElement>('infiniteScrollDetector')
 const shouldLoadMore = useElementVisibility(infiniteScrollDetectorElement)
 
-useHead({
-  title: $t('shelf_page_title'),
+useHead(() => {
+  const meta = []
+  if (isSomeoneElseBookshelf.value) {
+    meta.push({
+      name: 'robots',
+      content: 'noindex',
+    })
+  }
```

Comment:
We can safely set `noIndex` for all cases, since `/shelf` requires login if owned